Add Configuring the PDP and PCM and Driver Station Best Practices (#132)
wpilibsuite/frc-docs: source/docs/software/getting-started/driver-station-best-practices.rst

   c. Ports

      i. A built-in Ethernet port is highly preferred. Ensure that it's a full-sized port. The hinged Ethernet ports don't hold up to repeated use.
      ii. Use an Ethernet port saver to make your Ethernet connection. This extends the life of the port on the laptop. This is particularly important if you have a consumer-grade laptop with a hinged Ethernet port.
      iii. If the Ethernet port on your laptop is dodgy, either replace the laptop (recommended) or buy a USB Ethernet dongle from a reputable brand. Many teams find that USB Ethernet is less reliable than built-in Ethernet, primarily due to cheap hardware and bad drivers. The dongles given to rookies in the KOP have a reputation for working well.
      iv. 2 USB ports minimum

   d. A keyboard. It’s hard to quickly do troubleshooting on touch-only computers at the field.
   e. A solid-state disk (SSD). If the laptop has a rotating disk, spend $50 and replace it with a SSD.
   f. Updated to the current release of Windows 10. Windows 10 is the most common OS now seen at competitions, so bugs are more likely to be found than on Windows.

> 5. Install all Windows updates a week before the competition. This allows you time to ensure the updates will not interfere with driver station functions. To do so, open the Windows Updatesettings page and see that you're up-to-date. Install pending updates if not. Reboot and check again to make sure you’re up to date.

Reviewer: Missed a space
```suggestion
5. Install all Windows updates a week before the competition. This allows you time to ensure the updates will not interfere with driver station functions. To do so, open the Windows Update settings page and see that you're up-to-date. Install pending updates if not. Reboot and check again to make sure you’re up to date.
```

6. Change "Active Hours" for Windows Updates to prevent updates from installing during competition hours. Navigate to Start -> Settings -> Update & Security -> Windows Update, then select Change active hours. If you’re traveling to a competition, take time zone differences into account. This will help ensure your driver station does not reboot or fail due to update installing on the field.

> 7. Remove any 3rd party antivirus or antimalware software. Instead, use Windows Defender on Windows 10. Since you're only connecting to the internet for Windows and FRC software updating, the risk is low. Only install software on your driver station that's needed for driving.  Your goal here is to eliminate variables that might interfere with proper operation. Remove any unneeded preinstalled software ("crapware") that came with the machine. Don't use the laptop as your Steam machine for gaming back at the hotel the night before the event. Many teams go as far as having a separate programming laptop.

Reviewer: Extra space
```suggestion
7. Remove any 3rd party antivirus or antimalware software. Instead, use Windows Defender on Windows 10. Since you're only connecting to the internet for Windows and FRC software updating, the risk is low. Only install software on your driver station that's needed for driving. Your goal here is to eliminate variables that might interfere with proper operation. Remove any unneeded preinstalled software ("crapware") that came with the machine. Don't use the laptop as your Steam machine for gaming back at the hotel the night before the event. Many teams go as far as having a separate programming laptop.
```

8. Avoid managed Windows 10 installations from the school's IT department. These deployments are built for the school environment and often come with unwanted software that interferes with your robot's operation.
9. Laptop battery / power

   a. Turn off Put the computer to sleep in your power plan for both battery and powered operation.
   b. Turn off USB Selective Suspend:

>       i.Right click on the battery/charging icon in the tray, then select Power Options.

Reviewer: ```suggestion
      i. Right click on the battery/charging icon in the tray, then select Power Options.
```

>       ii.Edit the plan settings of your power plan.

Reviewer: ```suggestion
      ii. Edit the plan settings of your power plan.
```

>       iii.Click the Change advanced power settings link. iv.Scroll down in the advanced settings and disable the USB selective suspendsetting for both Battery and Plugged in.

Reviewer: ```suggestion
      iii. Click the Change advanced power settings link.
      iv. Scroll down in the advanced settings and disable the USB selective suspendsetting for both Battery and Plugged in.
```

   c. Ensure the laptop battery can hold a charge for at least an hour after making the changes above. This allows plenty of time for the robot and drive team to go through the queue and reach the alliance station without mains power.

10. Bring a trusted USB and Ethernet cable for use connecting to the roboRIO.
11. Add retention/strain relief to prevent your joystick/gamepad controllers from falling on the floor and/or yanking on the USB ports. This helps prevent issues with intermittent controller connections.
12. The Windows user account you use to drive must be a member of the Administrator group.

At the competition
------------------

1. Turn off Windows firewall using these instructions.
2. Turn off the Wi-Fi adapter, either using the dedicated hardware Wi-Fi switch or by disabling it in the Adapter Settings control panel.
3. Charge the driver station when it’s in the pit.

> 4. Disable / remove login passwords or ensure everyone on the drive team knows the password. You'd be surprised at how often drivers arrive at the field without knowing the password for the laptop.

Reviewer: ```suggestion
4. Remove login passwords or ensure everyone on the drive team knows the password. You'd be surprised at how often drivers arrive at the field without knowing the password for the laptop.
```

Driver Station Best Practices
=============================

> This document was created by Steve Peterson, with contributions from Juan Chong, James Cole-Henry, Rick Kosbab, Greg McKaskle, Chris Picone, Chris Roadfeldt, Joe Ross, and Ryan Sjostrand. The original post and followup posts can be found here: https://www.chiefdelphi.com/media/papers/3452 and a mirror of the document can be found `here. <https://media.screensteps.com/attachment_assets/assets/001/433/381/original/Driver_Station_Best_Practices_--_March_14_2018.pdf>`_

Reviewer 2: Replace the link to the screensteps. Just put a mirror of the copy in the "assets" directory relative to the location of this document.

> This document was created by Steve Peterson, with contributions from Juan Chong, James Cole-Henry, Rick Kosbab, Greg McKaskle, Chris Picone, Chris Roadfeldt, Joe Ross, and Ryan Sjostrand. The original post and followup posts can be found here: https://www.chiefdelphi.com/media/papers/3452 and a mirror of the document can be found `here. <https://media.screensteps.com/attachment_assets/assets/001/433/381/original/Driver_Station_Best_Practices_--_March_14_2018.pdf>`_

Reviewer 2: ```suggestion
This document was created by Steve Peterson, with contributions from Juan Chong, James Cole-Henry, Rick Kosbab, Greg McKaskle, Chris Picone, Chris Roadfeldt, Joe Ross, and Ryan Sjostrand. The original post and followup posts can be found `here <https://www.chiefdelphi.com/media/papers/3452>` and a mirror of the document can be found `here. <https://media.screensteps.com/attachment_assets/assets/001/433/381/original/Driver_Station_Best_Practices_--_March_14_2018.pdf>`_
```

Want to ensure the driver station isn't a stopper for your team at the FIRST Robotics Competition (FRC) field? Building and configuring a solid driver station laptop is an easy project for the time between stop build day and your competition. Read on to find lessons learned by many teams over thousands of matches.

> Prior to departing for the competition

Reviewer 2: ```suggestion
Prior To Departing For The Competition
```

9. Laptop battery / power

   a. Turn off Put the computer to sleep in your power plan for both battery and powered operation.
   b. Turn off USB Selective Suspend:

      i. Right click on the battery/charging icon in the tray, then select Power Options.
      ii. Edit the plan settings of your power plan.
      iii. Click the Change advanced power settings link.
      iv. Scroll down in the advanced settings and disable the USB selective suspendsetting for both Battery and Plugged in.

   c. Ensure the laptop battery can hold a charge for at least an hour after making the changes above. This allows plenty of time for the robot and drive team to go through the queue and reach the alliance station without mains power.

10. Bring a trusted USB and Ethernet cable for use connecting to the roboRIO.
11. Add retention/strain relief to prevent your joystick/gamepad controllers from falling on the floor and/or yanking on the USB ports. This helps prevent issues with intermittent controller connections.
12. The Windows user account you use to drive must be a member of the Administrator group.

> At the competition

Reviewer 2: ```suggestion
At The Competition
```

------------------

1. Turn off Windows firewall using these instructions.
2. Turn off the Wi-Fi adapter, either using the dedicated hardware Wi-Fi switch or by disabling it in the Adapter Settings control panel.
3. Charge the driver station when it’s in the pit.
4. Remove login passwords or ensure everyone on the drive team knows the password. You'd be surprised at how often drivers arrive at the field without knowing the password for the laptop.
5. Ensure your LabView code is deployed permanently and set to "run as startup", using the instructions in the LabView Tutorial. If you must deploy code every time you turn the robot on, you’re doing it wrong.
6. Limit web browsing to FRC related web sites. This minimizes the chance of getting malware during the competition.
7. Don't plan on using internet access to do software updates. There likely won't be any in the venue, and hotel Wi-Fi varies widely in quality. If you do need updates, contact a Control System Advisor in the pit.

> Before each match

Reviewer 2: ```suggestion
Before Each Match
```
